src/NuGetGallery.Core/Services/CloudBlobContainerWrapper.cs

```diff
@@ -40,10 +40,22 @@ public CloudBlobContainerWrapper(CloudBlobContainer blobContainer)
             return new BlobResultSegmentWrapper(segment);
         }
 
-        public Task CreateIfNotExistAsync()
+        public Task CreateIfNotExistAsync(BlobContainerPermissions permissions)
         {
+            var publicAccess = permissions?.PublicAccess;
+
+            ICancellableAsyncResult beginCreateIfNotExistsResult;
+            if (publicAccess.HasValue)
```

Comment:
can passed in `permissions` be `null`? I am assuming it could be given the check above.

perhaps consider adding a null check for `publicAccess` as well?

```suggestion
            if (publicAccess != null && publicAccess.HasValue)
```